nexus/test-utils/src/resource_helpers.rs

```diff
@@ -703,6 +704,53 @@ pub struct TestZpool {
     pub datasets: Vec<TestDataset>,
 }
 
+pub struct DiskTestBuilder<'a, N: NexusServer> {
```

Comment:
This builder specifically is cleaned up / used by default in https://github.com/oxidecomputer/omicron/pull/5939 . However, doing so in a "not-just-additive" way changes a lot of code unrelated to this PR, so, it got split out.